pkg/option/config.go

```diff
@@ -2976,6 +2982,15 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 		}
 	}
 
+	strictCIDR := vp.GetString(StrictModeCIDR)
+	if strictCIDR != "" {
+		c.StrictModeCIDR = cidr.MustParseCIDR(strictCIDR)
```

Comment:
I know this is taken from existing code to parse the native routing CIDR, but I think it would be if we could use `netip.Prefix` for `c.StrictModeCIDR` as the type here, we're trying to deprecate `net.IPNet` and the  `cidr.CIDR` types.